src/Collection/CollectionBuilder.php

```diff
@@ -363,6 +377,11 @@ protected function fixTask($task, $args)
         if ($task instanceof BuilderAwareInterface) {
             $task->setBuilder($this);
         }
+        // TODO: It would be better if we had a TaskIOInterface
+        // to go with the TaskIO trait. Perhaps in 2.0.
+        if (method_exists($task, 'setVerbosityThreshold')) {
```

Comment:
yes, this is a code smell. 
I think you should make OutputVerbosity interface and add it to BaseTask